validphys2/src/validphys/config.py

```diff
@@ -661,14 +676,38 @@ def produce_dataset(
 
         except LoadFailedError as e:
             raise ConfigError(e)
-
         if check_plotting:
             # normalize=True should check for more stuff
             get_info(ds, normalize=True)
             if not ds.commondata.plotfiles:
                 log.warning(f"Plotting files not found for: {ds}")
         return ds
 
+    def produce_t0dataset(
+        self,
+        *,
+        dataset_input,
+        theoryid,
+        cuts,
+        t0theoryid=None,
+        use_fitcommondata=False,
+        fit=None,
+        check_plotting: bool = False,
+    ):
+        """
+        Same as produce_dataset, but if a t0theoryid has been defined in the
+        runcard then those corresponding fktables will be linked.
+        """
+        ds = self.produce_dataset(
+            dataset_input=dataset_input,
+            theoryid=t0theoryid if t0theoryid else theoryid,
```

Comment:
Is it possible instead of doing the if-else here, for the parse rule to ensure that t0theoryid is equal to theoryid if not present?

If so I would prefer that.

Reply:
However, @scarlehoff's suggestion would make reading this part of the code less transparent. I honestly prefer Roy's solution, even if it is not "good-looking".

Reply:
This is what I wanted to do at first, but produce_ requires introducing another variable name - so it's t0theoryid in the runcard and as input to the production rule, but than the new variable should be .e.g produce_t0_theoryid. Which is done for example with t0pdfset and t0set, but I don't think is much better. 

parse_t0theoryid already requires the t0theoryid to exist, which is exactly not always the case

Since it only appears in two cases i went with the current solution, but I can do the first if you prefer

Reply:
Since the other case in which it appears is also t0, I think it makes sense to follow the same convention.

But :(
I think a nice addition to reportengine (cc @comane if you ever want to have a go at doing it) would be to level of parse rules so that a pase can depend on a previous parse. Maybe with a decorator `@priority(4)` or something like that.